### ISSUE TITLE
Add support for mac os 

### ISSUE THREAD

**Issue by camilamacedo86** [replicatedhq/kURL#3633]
## Description

Kurl currently only works on 

```
Supported operating systems

- Ubuntu 18.04
- Ubuntu 20.04 (Docker version >= 19.03.10)
- CentOS 7.4, 7.5, 7.6, 7.7, 7.8, 7.9, 8.1, 8.2, 8.3 (CentOS 8.x requires Containerd)
- RHEL 7.4, 7.5, 7.6, 7.7, 7.8, 7.9, 8.1, 8.2, 8.3, 8.4, 8.5, 8.6 (RHEL 8.x requires Containerd)
- Oracle Linux 7.4, 7.5, 7.6, 7.7, 7.8, 7.9, 8.1, 8.2, 8.3, 8.4, 8.5, 8.6 (OL 8.x requires Containerd)
- Amazon Linux 2
``` 

Therefore, would be great additional support for mac os X. 

## Motivations

- Allow users to be able to use kurl to test their app/releases on a remote cluster for example
- Allow users and contributors to use kurl in their dev local env without a need to use a container so that they can have a better experience and easily contribute with
- would be great to ensure homogeneity in all solutions under the org/replicate where all could work and be supported on max and Linux. 

------------

### Scenarios

**To install** 

1) Go to curl https://kurl.sh/
2) Try to run the default section `curl https://kurl.sh/latest | sudo bash`

** To install by following the getting started (step 4) **

Try to following item 4 from https://docs.replicated.com/vendor/tutorial-ui-install-app-manager
`curl -sSL https://k8s.kurl.sh/<your-app-name-and-channel> | sudo bash`

#### 1) error `sed: 1: "assets/Manifest": command a expects \ followed by text`

**Reason:** 
The reason for that is because of the commands `sed -i` used on the script, see (in this case):

``` 
    local filepath=
    filepath="$(package_filepath "${package}")"
    sed -i "/^$(printf '%s' "${package}").*/d" assets/Manifest # remove from manifest
    rm -f "${filepath}" # remove the file
```  

It occurs because [sed command with -i option failing on Mac, but works on Linux](https://stackoverflow.com/questions/4247068/sed-command-with-i-option-failing-on-mac-but-works-on-linux)

**Possible Solution(s)**

- replace:  sed -i "/^$(printf '%s' "${package}").*/d" assets/Manifest # remove from manifest
- with: `sed -i.bak "/^$(printf '%s' "${package}").*/d" assets/Manifest; rm -f .assets/Manifest*.bak`

OR 

-  Install :`brew install gnu-sed`

#### 2) Error `command md5sum not found`

- Reason: After solving `sed -i` command usage we will check errors with md5sum. The `md5sum` command used on the script does not work on mac by default.
- Possible Solution: we can install the command with `brew install md5sha1sum`

#### 3) Cannot executed binaries

Error:

```
./kurl.sh: line 1244: ./bin/kurl: cannot execute binary file
./kurl.sh: line 6734: ./bin/yamltobash: cannot execute binary file
```

**Comment by laverya** [comment]
kURL installs a significant number of packages, and relies on kubeadm, which is not supported on MacOS. 

And most of the on-mac k8s clusters I've found already include things like PVC provisioners and cluster networking.

I can understand the desire for a local development environment, but I'm not sure what value kURL would really provide in this scenario - particularly not for the amount of effort that would be required.

**Comment by camilamacedo86** [comment]
Hi @laverya,

Thank you for your time here. I totally agree that that would add value for development purposes only which probably brings it to a lower priority. Also, I think you are right.  `kubeadm` does not seems supported mac at all. 

So, thinking about the value that can brings vs effort it probably is not something valuable to get done  at all.  
So, I am totally fine with we close this one.

**Comment by camilamacedo86** [comment]
Hi @laverya,

Thank you for your reply.  

> kURL installs a significant number of packages, and relies on kubeadm, which is not supported on MacOS.

That is true. kubeadm would not run on mac. 

> I can understand the desire for a local development environment, but I'm not sure what value kURL would really provide in this scenario - particularly not for the amount of effort that would be required.

I understand your POV and go along with. +1. 
We can close this one one for sure.

**Comment by CpuID** [comment]
Future idea: support [colima](https://github.com/abiosoft/colima) as a MacOS solution in parallel to kubeadm for Linux based OS'es?

To have a play with it on either an Intel or M1 Mac:

```
brew install colima docker kubernetes-cli
colima start --kubernetes
kubectl get all -A
```

(there is a containerd option also, this uses docker as the container runtime by default)

```
~ ❯ kubectl get all -A                                                                                                                                                                          ⎈ colima 18:57:17
NAMESPACE     NAME                                          READY   STATUS    RESTARTS        AGE
kube-system   pod/coredns-597584b69b-r74k5                  1/1     Running   1 (5h57m ago)   6h14m
kube-system   pod/local-path-provisioner-79f67d76f8-phm94   1/1     Running   1 (5h57m ago)   6h14m
kube-system   pod/metrics-server-5c8978b444-5mxgb           1/1     Running   1 (5h57m ago)   6h14m

NAMESPACE     NAME                     TYPE        CLUSTER-IP      EXTERNAL-IP   PORT(S)                  AGE
default       service/kubernetes       ClusterIP   10.43.0.1       <none>        443/TCP                  6h14m
kube-system   service/kube-dns         ClusterIP   10.43.0.10      <none>        53/UDP,53/TCP,9153/TCP   6h14m
kube-system   service/metrics-server   ClusterIP   10.43.232.108   <none>        443/TCP                  6h14m

NAMESPACE     NAME                                     READY   UP-TO-DATE   AVAILABLE   AGE
kube-system   deployment.apps/coredns                  1/1     1            1           6h14m
kube-system   deployment.apps/local-path-provisioner   1/1     1            1           6h14m
kube-system   deployment.apps/metrics-server           1/1     1            1           6h14m

NAMESPACE     NAME                                                DESIRED   CURRENT   READY   AGE
kube-system   replicaset.apps/coredns-597584b69b                  1         1         1       6h14m
kube-system   replicaset.apps/local-path-provisioner-79f67d76f8   1         1         1       6h14m
kube-system   replicaset.apps/metrics-server-5c8978b444           1         1         1       6h14m
```

I can see the merits to `curl https://kurl.sh/latest | sudo bash` "just working" on either MacOS or Linux (copy paste to either yields a usable outcome effectively). Understood that it's an extra maintenance effort taking such a path though, which needs to be weighed up...

Worth noting: colima runs a native arm64 QEMU VM for containers on M1 Mac's, it does not do an emulated amd64 VM. So image architecture would need to be considered for everything...
